**Pre-Lecture**

In [2]:
import pandas as pd
url = "https://raw.githubusercontent.com/rfordatascience/tidytuesday/master/data/2020/2020-05-05/villagers.csv"
df = pd.read_csv(url)
df.isna().sum()

row_n           0
id              1
name            0
gender          0
species         0
birthday        0
personality     0
song           11
phrase          0
full_id         0
url             0
dtype: int64

In [3]:
import pandas as pd

url = "https://raw.githubusercontent.com/rfordatascience/tidytuesday/master/data/2020/2020-05-05/villagers.csv"
villagers_df = pd.read_csv(url)

# Get the number of rows and columns
rows, columns = villagers_df.shape
print(f"Rows: {rows}, Columns: {columns}")


Rows: 391, Columns: 11


2) 
Rows: 391, Columns: 11

Variables - Characteristics that are being measured

Observations - Individual datapoints measuring selected variables

In [4]:
import pandas as pd

# Load the dataset
url = "https://raw.githubusercontent.com/rfordatascience/tidytuesday/master/data/2020/2020-05-05/villagers.csv"
villagers_df = pd.read_csv(url)

# Get basic information about the dataset (e.g., column names, data types, non-null values)
print("Dataset Info:")
villagers_df.info()

# Get summary statistics for numerical columns (e.g., mean, std, min, max)
print("\nSummary Statistics for Numerical Columns:")
print(villagers_df.describe())

# Get a summary of all columns, including categorical ones
print("\nSummary Statistics for Categorical Columns:")
print(villagers_df.describe(include=['object']))

# You can also view the first few rows to get a sense of the dataset structure
print("\nFirst 5 Rows of the Dataset:")
print(villagers_df.head())


Dataset Info:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 391 entries, 0 to 390
Data columns (total 11 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   row_n        391 non-null    int64 
 1   id           390 non-null    object
 2   name         391 non-null    object
 3   gender       391 non-null    object
 4   species      391 non-null    object
 5   birthday     391 non-null    object
 6   personality  391 non-null    object
 7   song         380 non-null    object
 8   phrase       391 non-null    object
 9   full_id      391 non-null    object
 10  url          391 non-null    object
dtypes: int64(1), object(10)
memory usage: 33.7+ KB

Summary Statistics for Numerical Columns:
            row_n
count  391.000000
mean   239.902813
std    140.702672
min      2.000000
25%    117.500000
50%    240.000000
75%    363.500000
max    483.000000

Summary Statistics for Categorical Columns:
            id    name gender species birthday

5) 
An attribute, like df.shape is a characteristic or property of an object in python. When it is accessed, it does not affect the object.

A method, like df.describe(), is a function that can be performed on an object. They might take arguments, and after being called, it changes the object to which it corresponds.

**Post-Lecture**

6) 
From ChatGPT:

Summary:
count: Number of non-null entries.
mean: Average value.
std: Standard deviation (spread of data).
min: Smallest value.
25%: Value at the 25th percentile (first quartile).
50%: Median (middle value).
75%: Value at the 75th percentile (third quartile).
max: Largest value.


7) 
    1) df.dropna() 